apps/dtlaunch/app-b2.js

```diff
@@ -99,13 +99,31 @@
 
   Bangle.drawWidgets(); // To immediately update widget field to follow current theme - remove leftovers if previous app set custom theme.
   Bangle.loadWidgets();
-  drawPage(0);
+  drawPage(page);
 
-  for (let i = 4; i < apps.length; i++) { // Load the rest of the app icons that were not initially.
+  for (let i = 0; i < apps.length; i++) { // Load the rest of the app icons that were not initially.
+    if (i >= page*4 && i < Math.min(page*4+4, apps.length)) continue;
     if (apps[i].icon)
       apps[i].icon = s.read(apps[i].icon); // should just be a link to a memory area
   }
 
+  if (!Bangle.dtHandlePagePersist) {
+    Bangle.dtHandlePagePersist = (page) => {
+      // Function for persisting the active page when leaving dtlaunch.
+      if (page===undefined) {return this.page||0;}
```

Comment:
Here, `this` will be the global object, is that what we want?

Reply:
No that wasn't my intention. I thought I somehow attached the value to the function but I see that's not the case.